runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BatchGroupAlsoByWindowViaIteratorsFn.java

```diff
@@ -165,12 +168,17 @@ public WindowReiterable(
     }
 
     @Override
-    public Reiterator<V> iterator() {
+    public WindowReiterator<V> iterator() {
```

Comment:
It's odd that `ElementByteSizeObservableIterable::iterator` adds observers within the method body. I assume this is for historic reasons, since it doesn't seem to do anything now, and the comment documenting references a `setObserver` method that doesn't exist. Anyway, your change looks fine. But we should consider cleaning this up.

https://github.com/apache/beam/blob/6453e859badcb629ae2528b77d84235b7291ff89/sdks/java/core/src/main/java/org/apache/beam/sdk/util/common/ElementByteSizeObservableIterable.java#L49-L61

Reply:
I filed BEAM-9878 as a wish item.